ably/realtime_client.go

```diff
@@ -23,6 +23,12 @@ func NewRealtimeClient(opts *ClientOptions) (*RealtimeClient, error) {
 	if opts == nil {
 		panic("called NewRealtimeClient with nil ClientOptions")
 	}
+	_, error := opts.getFallbackHosts()
```

Comment:
I think we should consider this a general validation call on the options and hide the fact that validating includes checking fallbacks to be within that call, e.g.:

```go
if err := opts.validate(); err != nil {
	return nil, err
}
```

Also, call the error variable `err` to avoid shadowing the `error` type.

Reply:
If we do that, maybe it's worth moving the validation that happens on `newAuth` there as well.

Reply:
@tcard , I was looking into validations that can be moved to validate method from newAuth, but it seems newAuth contains its own conditional clauses embedded inside validation to determine host, token details, method, and clientID. Even if we move validations here, there will be a replication of logic in newAuth. I would suggest keeping validation simple for now.